apps/angularlogos/src/app/logos-list/logos-list.component.ts

```diff
@@ -1,26 +1,27 @@
-import { Component, OnInit } from '@angular/core';
+import { ChangeDetectionStrategy, Component } from '@angular/core';
 import { ActivatedRoute, Router } from '@angular/router';
-import { Observable, Subject, concat } from 'rxjs';
-import { debounceTime, startWith, switchMap, map, first, takeUntil, tap } from 'rxjs/operators';
+import { concat, Observable, Subject } from 'rxjs';
+import { debounceTime, first, map, startWith, switchMap, takeUntil, tap } from 'rxjs/operators';
 
 import { DataService } from '../shared/data.service';
 import { LogoEntry } from '../shared/logo-entry';
 
 @Component({
   selector: 'app-logos-list',
   templateUrl: './logos-list.component.html',
-  styleUrls: ['./logos-list.component.scss']
+  styleUrls: ['./logos-list.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class LogosListComponent implements OnInit {
-  logos$: Observable<LogoEntry[]>;
-  searchTerm$ = new Subject<string>();
-  firstSearchTerm$: Observable<string>;
+export class LogosListComponent {
+  readonly logos$: Observable<LogoEntry[]>;
+  readonly searchTerm$: Subject<string>;
+  readonly firstSearchTerm$: Observable<string>;
 
-  randomSeed = '?v=' + Math.floor(Math.random() * 10);
+  randomSeed: string = `?v=${Math.floor(Math.random() * 10)}`;
 
-  constructor(private ds: DataService, private route: ActivatedRoute, private router: Router) {}
+  constructor(private dataService: DataService, private route: ActivatedRoute, private router: Router) {
```

Comment:
Why did you move the init logic to the constructor? Even if it makes no difference here, IMO it's a golden rule that there should be nothing in the body of the cobstructor, unless you have a very good reason for it.

Reply:
If you declare variables with `readonly` you cannot initialise them in `ngOnInit`, only in the `constructor`.

Reply:
Aaah, I did not know that! LGTM then.
Do you think this is a good practice we should spread? Because it goes a bit against what we teach about ngOnInit

Reply:
Well.. the `readonly` practice is focused in particular on `Observables`. A colleague at work pointed it out to me. He read a blog post about why it has some benefits, will ask if he still can find it.

Since then I actually adopted that practice as well (when I don't forget).

A benefit of spreading it is that the declarations of your variables is very clear, just a name and a type. The actual assignment is being done in the `constructor` or `ngOnInit`. Certainly when you have a lot of variables are defined it can become a mess.

But since in Angular app you're working a lot with observables most of it is done within the `constructor` that is correct, and I know that it is even mentioned in the Angular docs to keep your `constructor` empty or as clean as possible.

But I think with this, it really boils down to preference, nothing more nothing less. Because I really can understand it looks weird and when I first saw it I had the same reaction, but after a while working with it I actually started liking it.